crates/wysiwyg/src/dom/nodes/dom_node.rs

```diff
@@ -492,7 +530,17 @@ impl DomNodeKind {
     }
 
     pub fn is_leaf_kind(&self) -> bool {
-        matches!(self, Self::Text | Self::LineBreak)
+        match self {
+            Self::Text | Self::LineBreak | Self::Mention => true,
+            Self::Generic
+            | Self::Formatting(_)
+            | Self::Link
+            | Self::ListItem
+            | Self::List
+            | Self::CodeBlock
+            | Self::Quote
+            | Self::Paragraph => false,
```

Comment:
In this case I think we'd better not use the wildcard because it makes adding a new leaf node more error prone (as it will default to being a non-leaf node).